From 7490711f4b6cb7298c7cbadbcb5c59f6ab4c13a2 Mon Sep 17 00:00:00 2001
From: Thiago Macieira <thiago.macieira@nokia.com>
Date: Mon, 20 Dec 2010 19:41:08 +0100
Subject: [PATCH 1/2] Add a way to obtain the DBusConnection* pointer from a QDBusConnection

This is internal API and the behaviour may change without notice. In
the future, I could use a different implementation without libdbus-1,
so this pointer may be something different. The use of DBusConnection
is also entirely implementation-defined, so keeping this connection
around is entirely unsupported.

You have been warned.
---
 src/dbus/qdbusconnection.cpp |   15 +++++++++++++++
 src/dbus/qdbusconnection.h   |    2 ++
 2 files changed, 17 insertions(+), 0 deletions(-)

Index: qt-everywhere-opensource-src-4.7.1/src/dbus/qdbusconnection.cpp
===================================================================
--- qt-everywhere-opensource-src-4.7.1.orig/src/dbus/qdbusconnection.cpp	2010-11-06 01:55:23.000000000 +0000
+++ qt-everywhere-opensource-src-4.7.1/src/dbus/qdbusconnection.cpp	2011-01-18 21:28:25.169196005 +0000
@@ -857,6 +857,21 @@
 }
 
 /*!
+    \internal
+    \since 4.8
+
+    Returns the internal, implementation-defined pointer for this
+    connection. Currently, this returns a DBusConnection* pointer,
+    without changing the reference count. It is the responsibility of
+    the caller to call dbus_connection_ref if it wants to store the
+    pointer.
+*/
+void *QDBusConnection::internalPointer() const
+{
+    return d ? d->connection : 0;
+}
+
+/*!
     Returns true if this QDBusConnection object is connected.
 */
 bool QDBusConnection::isConnected() const
Index: qt-everywhere-opensource-src-4.7.1/src/dbus/qdbusconnection.h
===================================================================
--- qt-everywhere-opensource-src-4.7.1.orig/src/dbus/qdbusconnection.h	2010-11-06 01:55:23.000000000 +0000
+++ qt-everywhere-opensource-src-4.7.1/src/dbus/qdbusconnection.h	2011-01-18 21:28:25.169196005 +0000
@@ -163,6 +163,8 @@
 
     QDBusConnectionInterface *interface() const;
 
+    void *internalPointer() const;
+
     static QDBusConnection connectToBus(BusType type, const QString &name);
     static QDBusConnection connectToBus(const QString &address, const QString &name);
     static void disconnectFromBus(const QString &name);
